### PR TITLE
SJQ2T-386: BE - Restore file endpoint

### PR DESCRIPTION
**Pull request by DraganMitrasinovicHtec**
Endpoint for restoring files, I only allowed restoring if the File is deleted and none of it's parents (including the bucket) are not.

### CODE REVIEW

File: backend/src/main/java/com/htecgroup/skynest/model/entity/FileMetadataEntity.java [hunk 1]
```diff
@@ -37,6 +37,10 @@ public boolean isDeleted() {
         || (parentFolder == null ? bucket.isDeleted() : parentFolder.isDeleted());
   }
 
+  public boolean someParentIsDeleted() {
```
Comment (ivanfhtec) [inline backend/src/main/java/com/htecgroup/skynest/model/entity/FileMetadataEntity.java:39]: you could call this in `isDeleted()` above

File: backend/src/main/java/com/htecgroup/skynest/service/impl/FileServiceImpl.java [hunk 2]
```diff
@@ -243,6 +243,34 @@ public void deleteFile(UUID fileId) {
...
 
+  @Override
+  @RecordAction(objectId = "[0].toString()", actionType = ActionType.RESTORE)
+  public FileResponse restoreFile(UUID fileId) {
+    LoggedUserDto currentUser = currentUserService.getLoggedUser();
+
+    FileMetadataEntity fileMetadataEntity =
+        fileMetadataRepository.findById(fileId).orElseThrow(FileNotFoundException::new);
+
+    if (!((ObjectEntity) fileMetadataEntity).isDeleted()) {
```
Comment (ivanfhtec) [inline backend/src/main/java/com/htecgroup/skynest/service/impl/FileServiceImpl.java]: this cast shouldn't be necessary

File: backend/src/main/java/com/htecgroup/skynest/model/entity/FileMetadataEntity.java (hunk 1 above)
Comment (ivanfhtec) [reply backend/src/main/java/com/htecgroup/skynest/model/entity/FileMetadataEntity.java:39]: we could do the same for folders

Comment (DraganMitrasinovicHtec) [reply backend/src/main/java/com/htecgroup/skynest/model/entity/FileMetadataEntity.java:39]: Added this

File: backend/src/main/java/com/htecgroup/skynest/service/impl/FileServiceImpl.java (hunk 2 above)
Comment (DraganMitrasinovicHtec) [reply backend/src/main/java/com/htecgroup/skynest/service/impl/FileServiceImpl.java]: Got my logic mixed up, ur right

**Review by DarkoVeljanoskiHtec — APPROVED** [review]
shinzou wo sasageyo !

**Pull request merged**